### PR TITLE
Add option to update dependencies after creating a release

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What changed?

Add option to update dependencies after creating a release

## Why?

Prevent having stale dependencies.

### CODE REVIEW

File: .github/workflows/create-tag.yml [hunk 1]
```diff
@@ -131,3 +135,13 @@ jobs:
 
           gh repo set-default ${{ github.repository }}
           gh release create "$TAG" --verify-tag --draft --title "$TAG" -F "$TEMPFILE"
+
+      - name: Create PR updating dependencies
+        if: ${{ github.event.inputs.update_deps == 'true' }}
+        run: |
+          make update-dependencies
```
Comment (yycptt) [inline .github/workflows/create-tag.yml:142]: probably also run `make go-generate` here?
I need to regenerate some mocks (common/archiver/s3store/mocks/generate.go) in https://github.com/temporalio/temporal/pull/5789

File: .github/workflows/create-tag.yml [hunk 2]
```diff
@@ -16,6 +16,10 @@ on:
       base_tag:
         description: "Base tag to generate commit list for release notes"
         required: false
+      update_deps:
+        type: boolean
+        description: "Create PR updatint dependencies post-release"
```
Comment (dnr) [inline .github/workflows/create-tag.yml:21]:
```suggestion
        description: "Create PR updating dependencies post-release"
```

File: .github/workflows/create-tag.yml [hunk 3]
```diff
@@ -131,3 +135,13 @@ jobs:
 
           gh repo set-default ${{ github.repository }}
           gh release create "$TAG" --verify-tag --draft --title "$TAG" -F "$TEMPFILE"
+
+      - name: Create PR updating dependencies
+        if: ${{ github.event.inputs.update_deps == 'true' }}
+        run: |
+          make update-dependencies
+          git checkout -b temporal-data/update-dependencies
```
Comment (dnr) [inline .github/workflows/create-tag.yml:143]: maybe it should use a unique branch name each time? if the branch isn't deleted from the last one the push will fail. or push with --force

Comment (rodrigozhou) [reply .github/workflows/create-tag.yml]: I added the base commit hash as part of branch to make it somewhat unique.

File: .github/workflows/create-tag.yml [hunk 4]
```diff
@@ -131,3 +135,14 @@ jobs:
...
+
+      - name: Create PR updating dependencies
+        if: ${{ github.event.inputs.update_deps == 'true' }}
+        run: |
+          make update-dependencies
+          make go-generate
+          git checkout -b temporal-data/update-dependencies-$(git rev-parse --short HEAD)
+          git add .
+          git commit -m "Update dependencies" --author ${{ github.actor }}
+          git push origin temporal-data/update-dependencies
```
Comment (dnr) [inline .github/workflows/create-tag.yml:147]: need to change this too?

File: .github/workflows/create-tag.yml (hunk 1 above)
Comment (dnr) [reply .github/workflows/create-tag.yml:142]: the other option would be to have update-dependencies depend on go-generate in the makefile. just mentioning, I think I like this better

File: .github/workflows/create-tag.yml (hunk 4 above)
Comment (rodrigozhou) [reply .github/workflows/create-tag.yml]: Fixed! Thanks

**Review by dnr — APPROVED** [review]

**Pull request merged**